sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java

```diff
@@ -0,0 +1,96 @@
+package org.sakaiproject.maven.plugin.component;
+ 
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+ 
+/**
+ * Traverses all the js files in sourceDir and copies them to targetDir, locating any js module import statements on the
+ * way and optionally adding the supplied query to the end. This is great for cache busting.
+ */
+@Mojo(name = "copyjs")
+public class CopyJSMojo extends AbstractMojo {
+
+  @Parameter
+  private String query;
+
+  @Parameter(required=true)
+  private String sourceDir;
+
+  @Parameter(required=true)
+  private String targetDir;
+
+  public void execute() throws MojoExecutionException {
+
+    try {
+      Files.walkFileTree(Paths.get(sourceDir), new SimpleFileVisitor<Path>() {
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+
+          Path currentTargetDir = Paths.get(targetDir).resolve(Paths.get(sourceDir).relativize(dir));
+          try {
+            Files.copy(dir, currentTargetDir);
+          } catch (FileAlreadyExistsException e) {
+            if (!Files.isDirectory(currentTargetDir)) throw e;
+          }
+          return FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+
+          if (file.toString().endsWith(".js")) {
```

Comment:
Do you think we'd also have to be concerned with .ts files?

Reply:
Don't think so. We've no plans around typescript at the moment. I mean, we could add it for future proofing sake, but I don't think it's necessary.

Reply:
Okay just checking, I guess it's not a big deal if we don't need it now.